src/Skia/Avalonia.Skia/GeometryImpl.cs

```diff
@@ -45,9 +45,12 @@ public bool FillContains(Point point)
         /// <inheritdoc />
         public bool StrokeContains(IPen? pen, Point point)
         {
-            _pathCache.UpdateIfNeeded(StrokePath, pen);
+            lock (this)
```

Comment:
We usually avoid `lock(this)` and introduce a separate `private object _lock = new();` field for locking purposes

Reply:
@kekekeks Updated. What are the advantages of having a separate lock object?

Reply:
@BAndysc unrelated project was pushed by accident.

> What are the advantages of having a separate lock object?

One could accidentally add lock on IGeometryImpl instance since it can be accessed from the outside, potentially dead-locking whole app.

Reply:
Oops, my bad, fixed. Thanks for the answer :)